src/Repository/TufValidatedComposerRepository.php

```diff
@@ -21,94 +15,23 @@
 
 class TufValidatedComposerRepository extends ComposerRepository
 {
-    /**
-     * @var bool
-     *   Indicates whether this Composer repository is validated by a parallel
-     *   TUF repository.
-     */
-    protected $isValidated;
-
-    /**
-     * @var array
-     *   Cache to short-circuit TUF operations when accessing the Composer root metadata.
-     *
-     *   Serves same purpose as parent::$rootData, but its visibility is private.
-     */
-    protected $tufValidatedRootData;
-
-    /**
-     * @var array
-     *   HTTP request options.
-     */
-    protected $options;
-
-    /**
-     * @var bool
-     *   Indicates whether the repository is expected to have a functioning http fallback.
-     *
-     *   Unlike ComposerRepository, this defaults to true if not set by the root composer.json.
-     *   HTTPS is not the primary means of security in TUF-validated repositories.
-     */
-    protected $allowSslDowngrade;
-
-    /**
-     * @var string
-     *   The URL to the composer repository root.
-     *
-     *   Serves same purpose as parent::$url, but its visibility is private.
-     */
-    protected $composerRepoUrl;
-
     /**
      * @var Updater
      */
     protected $tufRepo;
 
     /**
-     * @var EventDispatcher
-     */
-    protected $eventDispatcher;
-
-    /**
-     * @var HttpDownloader
-     */
-    protected $httpDownloader;
-
-    /**
-     * @var bool
-     */
-    protected $degradedMode = false;
-
-    /**
-     * @var IOInterface
+     * {@inheritDoc}
      */
-    protected $io;
-
     public function __construct(array $repoConfig, IOInterface $io, Config $config, HttpDownloader $httpDownloader, EventDispatcher $eventDispatcher = null)
     {
         parent::__construct($repoConfig, $io, $config, $httpDownloader, $eventDispatcher);
 
-        $this->isValidated = false;
-        $this->composerRepoUrl = $repoConfig['url'];
         if (!empty($repoConfig['tuf'])) {
-            $this->isValidated = true;
-            $this->eventDispatcher = $eventDispatcher;
-            $this->httpDownloader = $httpDownloader;
-            $this->io = $io;
-            if (!isset($repoConfig['options'])) {
-                $repoConfig['options'] = array();
-            }
-            $this->options = $repoConfig['options'];
-            $this->allowSslDowngrade = true;
-            if (array_key_exists('allow_ssl_downgrade', $repoConfig)) {
-                $this->allowSslDowngrade = $repoConfig['allow_ssl_downgrade'];
-            }
-
             $tufConfig = $repoConfig['tuf'];
 
             // @todo: Write a custom implementation of FileStorage that stores repo keys to user's global composer cache?
-            $allowList = 'a-z0-9.';
-            $repoPath = preg_replace('{[^'.$allowList.']}i', '-', $this->composerRepoUrl);
+            $repoPath = preg_replace('{[^a-z0-9.]}i', '-', $repoConfig['url']);
```

Comment:
I've never seen this `:alnum:` pattern before in regex! Can you by any chance link me to some documentation for it? I will add the inline comment either way, though.

Reply:
Another thought (that I'd rather run past @mbaynton first): could we not just use a SHA256 hash of the repo URL to create the directory name?

Reply:
`:alnum:` is a PCRE character class. [Here's the documentation](https://www.php.net/manual/en/regexp.reference.character-classes.php) including a list of them.

Reply:
Added the comment, but didn't change the regex because I'm not sure the one you're suggesting is valid. Unless I'm missing something, it contains an unclosed `]`...

Reply:
You're right: there _was_ an unclosed bracket (`[`). I'm not sure how that happened, but I've updated it to just escape the dot (`.`) instead. But it's no big deal either way.